modules/core/src/test/java/org/apache/synapse/core/axis2/ProxyServiceMessageReceiverTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.synapse.core.axis2;
+
+
+import org.apache.axiom.om.OMAbstractFactory;
+import org.apache.axiom.util.UIDGenerator;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.context.ConfigurationContext;
+import org.apache.axis2.context.OperationContext;
+import org.apache.axis2.context.ServiceContext;
+import org.apache.axis2.description.InOutAxisOperation;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+
+/**
+ * Unit tests to test methods in class ProxyServiceMessageReceiver
+ */
+public class ProxyServiceMessageReceiverTest {
+
+    /**
+     * Test message receipt by ProxyServiceMessageReceiver
+     *
+     * @throws Exception on a message receipt issue
+     */
+    @Test
+    public void testReceive() throws Exception {
+
+        //create ProxyServiceMessageReceiver instance
+        ProxyServiceMessageReceiver proxyServiceMessageReceiver = new ProxyServiceMessageReceiver();
+        proxyServiceMessageReceiver.setProxy(new ProxyService("TestProxy"));
+
+        MessageContext messageContext = createMessageContext();
+        Axis2SynapseEnvironment synapseEnvironment = getMockedSynapseEnvironment();
+
+        //set required values
+        MessageContextCreatorForAxis2.setSynConfig(new SynapseConfiguration());
+        MessageContextCreatorForAxis2.setSynEnv(synapseEnvironment);
+        messageContext.setEnvironment(synapseEnvironment);
+
+        //invoke
+        proxyServiceMessageReceiver.receive(((Axis2MessageContext) messageContext).getAxis2MessageContext());
```

Comment:
Don't we need to assert anything?

Reply:
It is returning nothing. So no need to assert.

Reply:
We shouldn't be asserting only the methods that are returning something. We need to find something to assert. May be a change in a field. Otherwise having the real method implementation empty could also pass the unit test.

Reply:
Done. Set a property in the inSequence and asserted it once returned from the method.